gax/src/test/java/com/google/api/gax/rpc/OperationCallableImplTest.java

```diff
@@ -0,0 +1,983 @@
+    assertThat(exception).isNotNull();
+    assertExceptionMatchesCode(statusCode, exception.getCause());
+    cause = (ApiException) exception.getCause();
+    assertThat(cause.getStatusCode()).isEqualTo(statusCode);
+    assertThat(future.getMetadata().isDone()).isTrue();
+    assertThat(future.getMetadata().isCancelled()).isFalse();
+  }
+
+  private void assertFutureCancelMetaCancel(OperationFuture<Color, Currency> future)
+      throws InterruptedException, ExecutionException, TimeoutException {
+    Exception exception = null;
+    try {
+      future.get(3, TimeUnit.SECONDS);
+    } catch (CancellationException e) {
+      exception = e;
+    }
+
+    assertThat(exception).isNotNull();
+    assertThat(future.isDone()).isTrue();
+    assertThat(future.isCancelled()).isTrue();
+
+    try {
+      future.peekMetadata().get();
+    } catch (CancellationException e) {
+      exception = e;
+    }
+    assertThat(future.peekMetadata()).isSameAs(future.peekMetadata());
+    assertThat(exception).isNotNull();
+    assertThat(future.peekMetadata().isDone()).isTrue();
+    assertThat(future.peekMetadata().isCancelled()).isTrue();
+
+    try {
+      future.getMetadata().get();
+    } catch (CancellationException e) {
+      exception = e;
+    }
+    assertThat(future.getMetadata()).isSameAs(future.getMetadata());
+    assertThat(exception).isNotNull();
+    assertThat(future.getMetadata().isDone()).isTrue();
+    assertThat(future.getMetadata().isCancelled()).isTrue();
+  }
+
+  private Color getColor(float blueValue) {
+    return new Color(0.0f, 0.0f, blueValue);
+  }
+
+  private ClientContext getClientContext(FakeChannel channel, ScheduledExecutorService executor) {
+    return ClientContext.newBuilder()
+        .setTransportChannel(FakeTransportChannel.of(channel))
+        .setExecutor(executor)
+        .build();
+  }
+
+  private OperationSnapshot getOperation(
+      String name, Object response, StatusCode errorCode, Object metadata, boolean done) {
+    FakeOperationSnapshot.Builder builder =
+        FakeOperationSnapshot.newBuilder().setName(name).setDone(done);
+    if (response != null) {
+      builder.setResponse(response);
+    }
+    if (errorCode != null) {
+      builder.setErrorCode(errorCode);
+    } else {
+      builder.setErrorCode(FakeStatusCode.of(Code.OK));
+    }
+    if (metadata != null) {
+      builder.setMetadata(metadata);
+    }
+    return builder.build();
+  }
+
+  private <RequestT> UnaryCallable<RequestT, OperationSnapshot> mockGetOpSnapshotCallable(
+      final Code returnStatusCode, final OperationSnapshot... results) {
+    return new UnaryCallable<RequestT, OperationSnapshot>() {
+      private int index = 0;
+
+      @Override
+      public ApiFuture<OperationSnapshot> futureCall(RequestT request, ApiCallContext context) {
+        OperationSnapshot response = results[index];
+        if (index < results.length - 1) {
+          index += 1;
```

Comment:
`+= 1`? Why not `++`?

Reply:
Personal preference: I like the uniformity of using `+=` for all values on the right hand side of the expression.